### PR TITLE
Update react docs links and deps

### PR DESCRIPTION
**Pull request by tnorling**
- Update doc links to point to dev
- Move storybook dependency to devDependencies

### CODE REVIEW

File: lib/msal-react/docs/getting-started.md [hunk 1]
```diff
@@ -69,7 +69,7 @@ export function App() {
 ### `useIsAuthenticated` hook
 
 As an alternative to the wrapper components above your app can use the `useIsAuthenticated` hook.
-You can read more about this hook in the [hooks doc](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/getting-started.md#useisauthenticated-hook)
+You can read more about this hook in the [hooks doc](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/hooks.md#useisauthenticated-hook)
```
Comment (jo-arroyo) [inline lib/msal-react/docs/getting-started.md:72]:
```suggestion
You can read more about this hook in the [hooks doc](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/hooks.md#useisauthenticated-hook).
```

File: lib/msal-react/docs/getting-started.md [hunk 2]
```diff
@@ -103,7 +103,7 @@ It will return the result of the login operation, any error that occurred and th
 
 Note: If you choose to use the redirect interaction type, this hook will only return if an error occurs. After the user signs in, the IDP will redirect the user back to your app at which point msal will reflect that the user has been signed in.
 
-You can read more about this hook in the [hooks doc](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/getting-started.md#usemsalauthentication-hook)
+You can read more about this hook in the [hooks doc](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/hooks.md#usemsalauthentication-hook)
```
Comment (jo-arroyo) [inline lib/msal-react/docs/getting-started.md:106]:
```suggestion
You can read more about this hook in the [hooks doc](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/hooks.md#usemsalauthentication-hook).
```

File: lib/msal-react/docs/getting-started.md [hunk 3]
```diff
@@ -134,7 +134,7 @@ Another way to trigger a sign-in is by using `msal-browser` APIs directly from t
 
 A hook that returns the `PublicClientApplication` instance, an array of all accounts currently signed in and an `inProgress` value that tells you what msal is currently doing.
 
-You can read more about this hook in the [hooks doc](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/getting-started.md#usemsal-hook)
+You can read more about this hook in the [hooks doc](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/hooks.md#usemsal-hook)
```
Comment (jo-arroyo) [inline lib/msal-react/docs/getting-started.md:137]:
```suggestion
You can read more about this hook in the [hooks doc](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-react/docs/hooks.md#usemsal-hook).
```

**Review by jo-arroyo — APPROVED** [review]
LGTM. Tiny nitpicks!

**Review by jasonnutter — APPROVED** [review]

**Pull request merged**